inst/include/interface/rcpp/rcpp_objects/rcpp_selectivity.hpp

```diff
@@ -43,16 +43,17 @@
  * */
 class SelectivityInterfaceBase : public FIMSRcppInterfaceBase {
  public:
-  static uint32_t id_g;
-  uint32_t id;
-  static std::map<uint32_t, SelectivityInterfaceBase*> selectivity_objects;
+  static uint32_t id_g; /**< static id of the recruitment interface base*/
+  uint32_t id;          /**< id of the recruitment interface base */
+  static std::map<uint32_t, SelectivityInterfaceBase*> selectivity_objects; /**< map associating the ids of SelectivityInterfaceBase to the objects */
 
   SelectivityInterfaceBase() {
     this->id = SelectivityInterfaceBase::id_g++;
     SelectivityInterfaceBase::selectivity_objects[this->id] = this;
     FIMSRcppInterfaceBase::fims_interface_objects.push_back(this);
   }
-
+  /** @brief get the ID of the interface base object
```

Comment:
is the below needed here analogous to rcpp_population.hpp?

virtual ~SelectivityInterfaceBase() {}

Reply:
Good catch, @timjmiller! I have added the destructor to the code.